File: app/src/main/java/in/woloo/www/subscribe/models/GetSubscriptionDetailsResponse.java
package in.woloo.www.subscribe.models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class GetSubscriptionDetailsResponse{

	@SerializedName("code")
	private int code;

	@SerializedName("data")
	private Data data;

	@SerializedName("message")
	private String message;

	@SerializedName("status")
	private String status;

	public int getCode(){
		return code;
	}

	public Data getData(){
		return data;
	}

	public String getMessage(){
		return message;
	}

	public String getStatus(){
		return status;
	}

	public class FutureSubscription{

		@SerializedName("image")
		private String image;

		@SerializedName("backgroud_color")
		private String backgroudColor;

		@SerializedName("price_with_gst")
		private String priceWithGst;

		@SerializedName("description")
		private String description;

		@SerializedName("discount")
		private String discount;

		@SerializedName("created_at")
		private String createdAt;

		@SerializedName("shield_color")
		private String shieldColor;

		@SerializedName("is_voucher")
		private int isVoucher;

		@SerializedName("deleted_at")
		private Object deletedAt;

		@SerializedName("frequency")
		private String frequency;

		@SerializedName("is_recommended")
		private int isRecommended;

		@SerializedName("is_expired")
		private int isExpired;

		@SerializedName("updated_at")
		private String updatedAt;

		@SerializedName("price")
		private String price;

		@SerializedName("name")
		private String name;

		@SerializedName("days")
		private int days;

		@SerializedName("currency")
		private String currency;

		@SerializedName("id")
		private int id;

		@SerializedName("before_discount_price")
		private int beforeDiscountPrice;

		@SerializedName("plan_id")
		private String planId;

		@SerializedName("status")
		private int status;

		@SerializedName("start_at")
		private String start_at;

		@SerializedName("end_at")
		private String end_at;

		public String getImage(){
			return image;
		}

		public String getBackgroudColor(){
			return backgroudColor;
		}

		public String getPriceWithGst(){
			return priceWithGst;
		}

		public String getDescription(){
			return description;
		}

		public String getDiscount(){
			return discount;
		}

		public String getCreatedAt(){
			return createdAt;
		}

		public String getShieldColor(){
			return shieldColor;
		}

		public int getIsVoucher(){
			return isVoucher;
		}

		public Object getDeletedAt(){
			return deletedAt;
		}

		public String getFrequency(){
			return frequency;
		}

		public int getIsRecommended(){
			return isRecommended;
		}

		public int getIsExpired(){
			return isExpired;
		}

		public String getUpdatedAt(){
			return updatedAt;
		}

		public String getPrice(){
			return price;
		}

		public String getName(){
			return name;
		}

		public int getDays(){
			return days;
		}

		public String getCurrency(){
			return currency;
		}

		public int getId(){
			return id;
		}

		public int getBeforeDiscountPrice(){
			return beforeDiscountPrice;
		}

		public String getPlanId(){
			return planId;
		}

		public int getStatus(){
			return status;
		}

		public String getStart_at() {
			return start_at;
		}

		public String getEnd_at() {
			return end_at;
		}
	}

	public class Data{

		@SerializedName("futureSubscription")
		private ArrayList<FutureSubscription> futureSubscription;

		@SerializedName("activeSubscription")
		private ActiveSubscription activeSubscription;

		@SerializedName("purchase_by")
		private String purchase_by;

		public String getPurchase_by() {
			return purchase_by;
		}

		public  ArrayList<FutureSubscription> getFutureSubscription(){
			return futureSubscription;
		}

		public ActiveSubscription getActiveSubscription(){
			return activeSubscription;
		}
	}

	public class ActiveSubscription{

		@SerializedName("image")
		private String image;

		@SerializedName("backgroud_color")
		private Object backgroudColor;

		@SerializedName("price_with_gst")
		private String priceWithGst;

		@SerializedName("description")
		private String description;

		@SerializedName("discount")
		private String discount;

		@SerializedName("created_at")
		private String createdAt;

		@SerializedName("shield_color")
		private Object shieldColor;

		@SerializedName("is_voucher")
		private int isVoucher;

		@SerializedName("deleted_at")
		private Object deletedAt;

		@SerializedName("frequency")
		private String frequency;

		@SerializedName("is_recommended")
		private int isRecommended;

		@SerializedName("is_expired")
		private int isExpired;

		@SerializedName("updated_at")
		private String updatedAt;

		@SerializedName("price")
		private String price;

		@SerializedName("name")
		private String name;

		@SerializedName("days")
		private String days;

		@SerializedName("currency")
		private String currency;

		@SerializedName("id")
		private int id;

		@SerializedName("before_discount_price")
		private Integer beforeDiscountPrice;

		@SerializedName("plan_id")
		private String planId;

		@SerializedName("corporate_name")
		private String corporate_name;

		@SerializedName("status")
		private int status;

		@SerializedName("start_at")
		private String start_at;

		@SerializedName("end_at")
		private String end_at;

		@SerializedName("gifted_by")
		private String gifted_by;

		public String getGifted_by() {
			return gifted_by;
		}

		public String getImage(){
			return image;
		}

		public Object getBackgroudColor(){
			return backgroudColor;
		}

		public String getPriceWithGst(){
			return priceWithGst;
		}

		public String getDescription(){
			return description;
		}

		public String getDiscount(){
			return discount;
		}

		public String getCreatedAt(){
			return createdAt;
		}

		public Object getShieldColor(){
			return shieldColor;
		}

		public int getIsVoucher(){
			return isVoucher;
		}

		public Object getDeletedAt(){
			return deletedAt;
		}

		public String getFrequency(){
			return frequency;
		}

		public int getIsRecommended(){
			return isRecommended;
		}

		public int getIsExpired(){
			return isExpired;
		}

		public String getUpdatedAt(){
			return updatedAt;
		}

		public String getPrice(){
			return price;
		}

		public String getName(){
			return name;
		}

		public String getDays(){
			return days;
		}

		public String getCurrency(){
			return currency;
		}

		public int getId(){
			return id;
		}

		public Integer getBeforeDiscountPrice(){
			return beforeDiscountPrice;
		}

		public String getPlanId(){
			return planId;
		}

		public int getStatus(){
			return status;
		}

		public String getStart_at() {
			return start_at;
		}

		public String getEnd_at() {
			return end_at;
		}

		public String getCorporate_name() {
			return corporate_name;
		}
	}
}

File: app/src/main/java/in/woloo/www/more/subscribe/models/GetSubscriptionDetailsResponse.kt
package `in`.woloo.www.more.subscribe.models

import com.google.gson.annotations.SerializedName

class GetSubscriptionDetailsResponse {
    @SerializedName("code")
    val code = 0

    @SerializedName("data")
    val data: Data? = null

    @SerializedName("message")
    val message: String? = null

    @SerializedName("status")
    val status: String? = null

    inner class FutureSubscription {
        @SerializedName("image")
        val image: String? = null

        @SerializedName("backgroud_color")
        val backgroudColor: String? = null

        @SerializedName("price_with_gst")
        val priceWithGst: String? = null

        @SerializedName("description")
        val description: String? = null

        @SerializedName("discount")
        val discount: String? = null

        @SerializedName("created_at")
        val createdAt: String? = null

        @SerializedName("shield_color")
        val shieldColor: String? = null

        @SerializedName("is_voucher")
        val isVoucher = 0

        @SerializedName("deleted_at")
        val deletedAt: Any? = null

        @SerializedName("frequency")
        val frequency: String? = null

        @SerializedName("is_recommended")
        val isRecommended = 0

        @SerializedName("is_expired")
        val isExpired = 0

        @SerializedName("updated_at")
        val updatedAt: String? = null

        @SerializedName("price")
        val price: String? = null

        @SerializedName("name")
        val name: String? = null

        @SerializedName("days")
        val days = 0

        @SerializedName("currency")
        val currency: String? = null

        @SerializedName("id")
        val id = 0

        @SerializedName("before_discount_price")
        val beforeDiscountPrice = 0

        @SerializedName("plan_id")
        val planId: String? = null

        @SerializedName("status")
        val status = 0

        @SerializedName("start_at")
        val start_at: String? = null

        @SerializedName("end_at")
        val end_at: String? = null
    }

    inner class Data {
        @SerializedName("futureSubscription")
        val futureSubscription: ArrayList<FutureSubscription>? = null

        @SerializedName("activeSubscription")
        val activeSubscription: ActiveSubscription? = null

        @SerializedName("purchase_by")
        val purchase_by: String? = null
    }

    inner class ActiveSubscription {
        @SerializedName("image")
        val image: String? = null

        @SerializedName("backgroud_color")
        val backgroudColor: Any? = null

        @SerializedName("price_with_gst")
        val priceWithGst: String? = null

        @SerializedName("description")
        val description: String? = null

        @SerializedName("discount")
        val discount: String? = null

        @SerializedName("created_at")
        val createdAt: String? = null

        @SerializedName("shield_color")
        val shieldColor: Any? = null

        @SerializedName("is_voucher")
        val isVoucher = 0

        @SerializedName("deleted_at")
        val deletedAt: Any? = null

        @SerializedName("frequency")
        val frequency: String? = null

        @SerializedName("is_recommended")
        val isRecommended = 0

        @SerializedName("is_expired")
        val isExpired = 0

        @SerializedName("updated_at")
        val updatedAt: String? = null

        @SerializedName("price")
        val price: String? = null

        @SerializedName("name")
        val name: String? = null

        @SerializedName("days")
        val days: String? = null

        @SerializedName("currency")
        val currency: String? = null

        @SerializedName("id")
        val id = 0

        @SerializedName("before_discount_price")
        val beforeDiscountPrice: Int? = null

        @SerializedName("plan_id")
        val planId: String? = null

        @SerializedName("corporate_name")
        val corporate_name: String? = null

        @SerializedName("status")
        val status = 0

        @SerializedName("start_at")
        val start_at: String? = null

        @SerializedName("end_at")
        val end_at: String? = null

        @SerializedName("gifted_by")
        val gifted_by: String? = null
    }
}